### DISCUSSION TITLE
Deploy Garnet to On-Prem Production Server

### DISCUSSION THREAD

**Discussion by Akinnagbe** [microsoft/garnet#658] [Q&A]
I want to know how to deploy garnet to an on-prem production server. A step-by-step guide would be helpful.

Thanks in advance.

**Comment by wessisaac** [comment]
In Windows OS, 
1. download 64 bit package from Git-Hub release (https://github.com/microsoft/garnet/releases)
2. Place the downloaded package in your convenient place. Eg: C:\Program Files\Garnet
3. In windows, Garnet as a service is best. To install garnet as a service, download 64 bit nssm from the url "https://nssm.cc/download" and place the nssm.exe folder into your Garnet package folder
4. In the command prompt enter the command, >nssm install Garnet
5. A GUI will open, in the path text box you can browse your GarnetServer.exe path
      Path: 	C:\Program Files\Garnet\GarnetServer.exe
6. In the argument text box, please enter the below command with path of your Garnet configuration file
     Arguments:	GarnetServer.exe --config-import-path "C:\Program Files\Garnet\Garnet.conf"
7. Then click install button
Now you can access Garnet in windows 
To use redis-cli in windows, download redis-windows package from the path "https://github.com/zkteco-home/redis-windows/releases/tag/7.4.2", then copy and paste the redis-cli.exe file (redis-clie.exe single file is enough) in Garnet package folder.
In command prompt, you can place the Garnet path, then redis-cli commands can be used.
